File: frontend/types/patient.ts
export type Gender = "male" | "female" | "other" | "unknown";

export type ProgramType =
  | "Public Health Enrollment"
  | "Primary Care"
  | "Chronic Disease Program"
  | "Virtual Care"
  | "Claims Support";

export interface Patient {
  id: string;
  firstName: string;
  lastName: string;
  gender: Gender;
  birthDate: string;
  phone: string;
  email: string;
  province: string;
  healthCardNumber: string;
  programType: ProgramType;
  registeredDate: string;
}

File: backend/app/models/patient.py
from pydantic import BaseModel
from typing import Literal


class PatientCreate(BaseModel):
    firstName: str
    lastName: str
    birthDate: str
    gender: Literal["male", "female", "other", "unknown"]
    phone: str
    email: str
    address: str
    province: str
    healthCardNumber: str
    programType: str
    consent: bool


class PatientResponse(BaseModel):
    id: str
    firstName: str
    lastName: str
    birthDate: str
    gender: str
    phone: str
    email: str
    address: str
    province: str
    healthCardNumber: str
    programType: str
    registeredDate: str